src/topp/FileConverter.cpp

```diff
@@ -452,6 +526,27 @@ class TOPPFileConverter :
       if (fm.size() > 0) EDTAFile().store(out, fm);
       else if (cm.size() > 0) EDTAFile().store(out, cm);
     }
+    else if (out_type == FileTypes::CACHEDMZML)
+    {
+      // Determine output path for meta information (empty mzML)
+      std::vector<String> split_out;
```

Comment:
there is a considerable bit of code duplication when handling the paired files.
Maybe you can find a way to put that into a function?